core/coreunix/add.go

```diff
@@ -68,12 +68,14 @@ type Adder struct {
 	Out        chan<- interface{}
 	Progress   bool
 	Hidden     bool
+	TopHidden  bool
 	Pin        bool
 	Trickle    bool
 	RawLeaves  bool
 	Silent     bool
 	Wrap       bool
 	Name       string
+	BaseName   string
```

Comment:
Has this been pushed to the interface?

Reply:
I'm also not sure if we really need to wire this all the way through. If we're _just_ wiring this through to name the event, we can probably apply the name in the add command (that is, it's a concern of the current HTTP API, not really of the core API).

Reply:
The end goal is to not have this. I did this only to see if the main change would fundamentally break the `ipfs add` logic.

Reply:
Got it.